File: backend/internal/domain/project.go
package domain

import "time"

// Project はJiraプロジェクト情報を表すエンティティ
type Project struct {
	ID              int64     `db:"id" json:"id"`
	JiraProjectID   string    `db:"jira_project_id" json:"jira_project_id"`
	Key             string    `db:"key" json:"key"`
	Name            string    `db:"name" json:"name"`
	LeadAccountID   *string   `db:"lead_account_id" json:"lead_account_id"`
	LeadEmail       *string   `db:"lead_email" json:"lead_email"`
	OrganizationID  *int64    `db:"organization_id" json:"organization_id"`
	CreatedAt       time.Time `db:"created_at" json:"created_at"`
	UpdatedAt       time.Time `db:"updated_at" json:"updated_at"`
}

// ProjectWithStats はプロジェクトと統計情報
type ProjectWithStats struct {
	Project
	TotalIssues   int `json:"total_issues"`
	RedIssues     int `json:"red_issues"`
	YellowIssues  int `json:"yellow_issues"`
	GreenIssues   int `json:"green_issues"`
	OpenIssues    int `json:"open_issues"`
	DoneIssues    int `json:"done_issues"`
}

File: frontend/src/types/project.ts
export type DelayStatus = 'RED' | 'YELLOW' | 'GREEN'

export interface Project {
  id: number
  jira_project_id: string
  key: string
  name: string
  lead_account_id: string | null
  lead_email: string | null
  organization_id: number | null
  created_at: string
  updated_at: string
  red_count: number
  yellow_count: number
  green_count: number
  open_count: number
  total_count: number
  delay_status: DelayStatus
}

export interface PaginationMeta {
  page: number
  per_page: number
  total: number
  total_pages: number
}

export interface ProjectListResponse {
  data: Project[]
  pagination: PaginationMeta
}

export type SortOption = 'name' | 'name_desc' | 'delay_count'

export type DelayFilter = 'ALL' | DelayStatus
